libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton/AttributeFilterButtonUtils.ts

```diff
@@ -0,0 +1,217 @@
+    isAttributeElementsByRef,
+    isNegativeAttributeFilter,
+    newNegativeAttributeFilter,
+    newPositiveAttributeFilter,
+    ObjRef,
+} from "@gooddata/sdk-model";
+import { UseCancelablePromiseStatus } from "@gooddata/sdk-ui";
+import isEmpty from "lodash/isEmpty";
+import { IntlShape } from "react-intl";
+import isNil from "lodash/isNil";
+
+export const getInitialSelectedOptions = (currentFilter: IAttributeFilter): IAttributeElement[] => {
+    // the as any cast is ok here, the data will get fixed once the element load completes
+    // this serves only to have some initial state here so that when full element data is loaded
+    // it automatically sets the props.filter.elements as selected
+    return currentFilter
+        ? (attributeElementsToAttributeElementArray(filterAttributeElements(currentFilter)) as any)
+        : [];
+};
+
+export const getInitialIsInverted = (currentFilter: IAttributeFilter): boolean => {
+    return currentFilter ? isNegativeAttributeFilter(currentFilter) : true;
+};
+
+export const isCancelablePromisePending = (cancelablePromiseStatus: UseCancelablePromiseStatus) => {
+    return cancelablePromiseStatus === "pending";
+};
+
+export const isCancelablePromiseLoading = (cancelablePromiseStatus: UseCancelablePromiseStatus) => {
+    return cancelablePromiseStatus === "loading";
+};
+
+export const isCancelablePromisePendingOrLoading = (cancelablePromiseStatus: UseCancelablePromiseStatus) => {
+    return (
+        isCancelablePromiseLoading(cancelablePromiseStatus) ||
+        isCancelablePromisePending(cancelablePromiseStatus)
+    );
+};
+
+export const getBackend = (backend: IAnalyticalBackend, props: any) => {
+    return backend.withTelemetry("AttributeFilter", props);
+};
+
+export const prepareElementsQuery = (
+    backend: IAnalyticalBackend,
+    workspace: string,
+    filterObjRef: ObjRef,
+    parentFilters: IAttributeFilter[],
+    parentFilterOverAttribute: ObjRef | ((parentFilter: IAttributeFilter, index: number) => ObjRef),
+    offset: number,
+    limit: number,
+    searchQuery: string,
+) => {
+    const preparedElementQuery = backend
+        .workspace(workspace)
+        .attributes()
+        .elements()
+        .forDisplayForm(filterObjRef)
+        .withOptions({
+            ...(!isEmpty(searchQuery) ? { filter: searchQuery } : {}),
+        })
+        .withOffset(offset)
+        .withLimit(limit);
+
+    if (isParentFilteringEnabled(backend)) {
+        if (parentFilters && !isParentFiltersElementsByRef(parentFilters)) {
+            // eslint-disable-next-line no-console
+            console.error("Parent filters must be defined by uris to enable parent-child filtering feature");
+        } else {
+            preparedElementQuery.withAttributeFilters(
+                getValidElementsFilters(parentFilters, parentFilterOverAttribute),
+            );
+        }
+    }
+
+    return preparedElementQuery;
+};
+
+export const prepareElementsTitleQuery = (
```

Comment:
Should we put this as a local function to `useFetchInitialElements.ts`? (it's not used anywhere else)

Reply:
The function moved to the hook